src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,431 @@
+            end
+          end
+        end
+
+        clipped = clipped.left
+        if clipped.left == startclipped.left
+          break
+        end
+      end
+    end
+
+    clipping = clipping.left
+    if clipping.left == startclipping.left
+      break
+    end
+  end
+
+  collectedrings = collectclipped(entering)
+
+  # When no interesction have been registered with any of the rings, the clipping ring either
+  # encircles everything or is completely contained in the clipped polygon.
+  if !any(intersected)
+    o = orientation(ring)
+    allcontained = all(v ∈ PolyArea(ring) for v in vertices(polyrings[1]))
+    if (o == CCW && allcontained)
+      return poly
+    end
+    if (o == CW && !allcontained)
+      push!(collectedrings, polyrings[1])
+      intersected[1] = true
+      for polyring in polyrings[2:end]
+        if !all(v ∈ PolyArea(ring) for v in vertices(polyring))
+          push!(collectedrings, polyring)
+        end
+      end
+      if any(v ∈ PolyArea(polyrings[1]) for v in vertices(ring))
+        push!(collectedrings, ring)
+      end
+      return PolyArea(collectedrings...)
+    end
+    if all(v ∈ PolyArea(polyrings[1]) for v in vertices(ring))
+      push!(collectedrings, ring)
+    end
+  end
+
+  # Handle all the non-intersected rings. Add them if they are contained in the clipping ring.
+  polys = PolyArea[]
+  for r in collectedrings
+    newpolyrings = [r]
+    valid = true
+    for k in eachindex(intersected)
+      if !intersected[k]
+        # Check if majority of vertices are inside the clipping ring.
+        ins = count(v ∈ PolyArea(ring) for v in vertices(polyrings[k]))
+        if orientation(ring) == CCW
+          if ins > (length(vertices(polyrings[k])) - ins)
+            if orientation(polyrings[k]) == CCW
+              pushfirst!(newpolyrings, polyrings[k])
+            else
+              push!(newpolyrings, polyrings[k])
+            end
+            intersected[k] = true
+          elseif vertices(ring)[1] ∈ PolyArea(polyrings[k]) && k > 1
+            # If the ring is contained within one of the inner rings, invalidate it.
+            valid = false
+          end
+        else
+          if ins < (length(vertices(polyrings[k])) - ins)
+            if orientation(polyrings[k]) == CCW
+              pushfirst!(newpolyrings, polyrings[k])
+            else
+              push!(newpolyrings, polyrings[k])
+            end
+            intersected[k] = true
+          end
+        end
+      end
+    end
+    if valid
+      push!(polys, PolyArea(newpolyrings))
+    end
+  end
+
+  n = length(polys)
+  n == 0 ? nothing : (n == 1 ? polys[1] : GeometrySet(polys))
+end
+
+clip(poly::Polygon, other::Geometry, method::WeilerAthertonClipping) = clip(poly, boundary(other), method)
+
+function clip(poly::Polygon, multi::Multi, method::WeilerAthertonClipping)
+  for r in parent(multi)
+    poly = clip(poly, r, method)
+    if isnothing(poly)
+      return nothing
+    end
+  end
+  poly
+end
```

Comment:
Same here. This method is not specific to WeilerAtherton. In order to facilitate the review, please remove it.